src/internal/serviceenv/service_env.go

```diff
@@ -32,6 +33,8 @@ const clusterIDKey = "cluster-id"
 // create more, if they want to create multiple pachyderm "clusters" served in
 // separate goroutines.
 type ServiceEnv interface {
+	AuthServer() auth_server.APIServer
```

Comment:
Add accessors for services to ServiceEnv - this replaces the cross-service references in TransactionEnv.